ChangeSet
  1.1898 04/04/21 16:45:18 chrisw@osdl.org +1 -0
  [IPV4]: Fix return value on MCAST_MSFILTER error case.

  net/ipv4/ip_sockglue.c
    1.23 04/04/21 16:45:03 chrisw@osdl.org +1 -1
    [IPV4]: Fix return value on MCAST_MSFILTER error case.

diff -Nru a/net/ipv4/ip_sockglue.c b/net/ipv4/ip_sockglue.c
--- a/net/ipv4/ip_sockglue.c	Fri Apr 23 06:49:48 2004
+++ b/net/ipv4/ip_sockglue.c	Fri Apr 23 06:49:48 2004
@@ -800,7 +800,7 @@
 				goto mc_msf_out;
 			}
 			if (GROUP_FILTER_SIZE(gsf->gf_numsrc) > optlen) {
-				err = EINVAL;
+				err = -EINVAL;
 				goto mc_msf_out;
 			}
 			msize = IP_MSFILTER_SIZE(gsf->gf_numsrc);
.........................................................................
# vim: syntax=diff

